src/eval.rs

```diff
@@ -75,8 +75,15 @@ pub fn create_ecx<'mir, 'tcx: 'mir>(
             config.tracked_pointer_tag,
             config.tracked_alloc_id,
         ),
-    );
+        );
     // Complete initialization.
+
+    // Initialize the `environ` static
+    let layout = ecx.layout_of(ecx.tcx.types.usize)?;
+    let place = ecx.allocate(layout, MiriMemoryKind::Machine.into());
+    ecx.write_scalar(Scalar::from_machine_usize(0, &*ecx.tcx), place.into())?;
+    ecx.memory.extra.environ = Some(place);
```

Comment:
This is way too specific to be out here in `eval.rs`.

What I was thinking is `EnvVars::init` should do it. You can move that up above `init_extern_statics`.